[All Hosts] (ux) Clarify distinction between various 'auto-' features (#5868)
OfficeDev/office-js-docs-pr: docs/develop/run-code-on-document-open.md

---
title: Configure a document to run code when it opens
description: Learn how to configure an Office file to run code when the document opens.
ms.topic: how-to
ms.date: 09/01/2026
ms.localizationpriority: medium
---

# Configure a document to run code when it opens

You can configure your Office Add-in to load and run code as soon as the document is opened. This is useful if you need to register event handlers, pre-load data for the task pane, synchronize UI, or perform other tasks before the add-in is visible.

> [!TIP]

> > This article describes a feature that enables an add-in to programmatically configure itself to run code when a document opens. The technique has *document scope*, meaning that it must be applied to each document individually. This feature should be kept distinct from three similar features:

Reviewer: ```suggestion
> This article describes a feature that enables an add-in to programmatically configure itself to run code when a document opens. The technique has *document scope*, meaning that it must be applied to each document individually. This feature is distinct from three similar features:
```
